skimage/feature/corner.py

```diff
@@ -90,16 +95,24 @@ def structure_tensor(image, sigma=1, mode='constant', cval=0):
 
     """
 
-    image = _prepare_grayscale_input_2D(image)
+    # image = _prepare_grayscale_input_2D(image)
 
-    imx, imy = _compute_derivatives(image, mode=mode, cval=cval)
+    # imx, imy = _compute_derivatives(image, mode=mode, cval=cval)
+
+    # # structure tensore
+    # Axx = ndi.gaussian_filter(imx * imx, sigma, mode=mode, cval=cval)
+    # Axy = ndi.gaussian_filter(imx * imy, sigma, mode=mode, cval=cval)
+    # Ayy = ndi.gaussian_filter(imy * imy, sigma, mode=mode, cval=cval)
 
-    # structure tensore
-    Axx = ndi.gaussian_filter(imx * imx, sigma, mode=mode, cval=cval)
-    Axy = ndi.gaussian_filter(imx * imy, sigma, mode=mode, cval=cval)
-    Ayy = ndi.gaussian_filter(imy * imy, sigma, mode=mode, cval=cval)
+    # return Axx, Axy, Ayy
 
-    return Axx, Axy, Ayy
+    out = ()
+    derivatives = list(_compute_derivatives(image, mode=mode, cval=cval))
+    for i in range(len(derivatives)):
+        for j in range(len(derivatives)):
```

Comment:
You can just use `range(i, len(derivatives))` here and avoid the `if` statement on the next line. =)